src/AWS.Deploy.CLI/ConsoleUtilities.cs

```diff
@@ -306,6 +307,68 @@ public string AskUserForValue(string message, string defaultValue, bool allowEmp
             return userValue;
         }
 
+        public Dictionary<string, string> AskUserForKeyValue(Dictionary<string, string> keyValue)
+        {
+            keyValue ??= new Dictionary<string, string>();
+
+            if (keyValue.Keys.Count == 0)
+            {
+                AskToAddKeyValuePair(keyValue);
+                return keyValue;
+            }
+
+            const string ADD = "Add new environment variable";
+            const string UPDATE = "Update existing environment variable";
+            const string DELETE = "Delete environment variable";
+            var operations = new List<string> { ADD, UPDATE, DELETE };
+
+            var selectedOperation = AskUserToChoose(operations, "Select which operation you want to perform:", ADD);
+
+            if(selectedOperation.Equals(ADD))
+                AskToAddKeyValuePair(keyValue);
+
+            if(selectedOperation.Equals(UPDATE))
+                AskToUpdateKeyValuePair(keyValue);
+
+            if(selectedOperation.Equals(DELETE))
+                AskToDeleteKeyValuePair(keyValue);
+
+            return keyValue;
+        }
+
+        private void AskToAddKeyValuePair(Dictionary<string, string> keyValue)
+        {
+            const string RESET = "<reset>";
+            var variableName = string.Empty;
+            while (string.IsNullOrEmpty(variableName))
+            {
+                _interactiveService.WriteLine("Enter the environment variable name:");
+                variableName = _interactiveService.ReadLine()?.Trim() ?? "";
+            }
+
+            _interactiveService.WriteLine($"Enter the environment variable value (type {RESET} to reset):");
+            var variableValue = _interactiveService.ReadLine()?.Trim() ?? "";
+            if (string.Equals(RESET, variableValue.Trim(), StringComparison.OrdinalIgnoreCase) ||
+                string.Equals($"'{RESET}'", variableValue.Trim(), StringComparison.OrdinalIgnoreCase))
+                variableValue = keyValue.ContainsKey(variableName) ? keyValue[variableName] : "";
+
+            keyValue[variableName] = variableValue;
+        }
+
+        private void AskToUpdateKeyValuePair(Dictionary<string, string> keyValue)
+        {
+            var selectedKey = AskUserToChoose(keyValue.Keys.ToList(), "Select the environment variable you wish to update:", null);
+            var selectedValue = AskUserForValue("Enter the environment variable value:", keyValue[selectedKey], true);
+
+            keyValue[selectedKey] = selectedValue;
+        }
+
+        private void AskToDeleteKeyValuePair(Dictionary<string, string> keyValue)
+        {
+            var selectedKey = AskUserToChoose(keyValue.Keys.ToList(), "Select the environment variable you wish to delete:", null);

```

Comment:
Refer to the above comment on message prompts.

Reply:
Done